apps/settings/js/modules/bluetooth/bluetooth_connection_manager.js

```diff
@@ -0,0 +1,674 @@
+     */
+    connectedAddress: null,
+
+    /**
+     * The connected profiles of device that we are connected.
+     *
+     * @public
+     * @memberOf BluetoothConnectionManager
+     * @type {String}
+     */
+    connectedProfiles: null,
+
+    /**
+     * Default adapter of Bluetooth.
+     *
+     * @access private
+     * @memberOf BluetoothConnectionManager
+     * @type {Object BluetoothAdapter}
+     */
+    _defaultAdapter: null,
+
+    /**
+     * Init BluetoothConnectionManager module.
+     *
+     * @access private
+     * @memberOf BluetoothConnectionManager
+     */
+    _init: function btcm__init() {
+      // Observe 'defaultAdapter' property for reaching default adapter.
+      AdapterManager.observe('defaultAdapter',
+                             this._onDefaultAdapterChanged.bind(this));
+      this._onDefaultAdapterChanged(AdapterManager.defaultAdapter);
+      this._restoreConnection();
+    },
+
+    /**
+     * Init observable properties from default adapter.
+     *
+     * @access private
+     * @memberOf BluetoothConnectionManager
+     * @param {Object BluetoothAdapter} adapter
+     */
+    _initProperties: function btcm__initProperties(adapter) {
+      // Update connection status and profile
+      this.getConnectedDevices().then((connectedDeviceItemsMap) => {
+        for (var address in connectedDeviceItemsMap) {
+          this.connectedAddress = address;
+          this.connectedProfiles =
+            connectedDeviceItemsMap[address].connectedProfiles;
+          Debug('_initProperties(): this.connectedAddress = ' +
+                this.connectedAddress);
+          Debug('_initProperties(): this.connectedProfiles = ' +
+                JSON.stringify(this.connectedProfiles));
+        }
+      }, (reason) => {
+        Debug('_initProperties(): rejected in getConnectedDevices');
+      });
+    },
+
+    /**
+     * Only reset properties since there is no available default adapter.
+     *
+     * @access private
+     * @memberOf BluetoothConnectionManager
+     */
+    _resetProperties: function btcm__resetProperties() {
+      // Reset connection status.
+      this.connectedAddress = null;
+      this.connectedProfiles = null;
+    },
+
+    /**
+     * 'defaultAdapter' change event handler from adapter manager for
+     * updating adapter immediately.
+     *
+     * @access private
+     * @memberOf BluetoothConnectionManager
+     * @param {Object BluetoothAdapter} newAdapter
+     * @param {Object BluetoothAdapter} oldAdapter
+     */
+    _onDefaultAdapterChanged:
+    function btcm__onDefaultAdapterChanged(newAdapter, oldAdapter) {
+      Debug('_onDefaultAdapterChanged(): newAdapter = ' + newAdapter);
+      Debug('_onDefaultAdapterChanged(): oldAdapter = ' + oldAdapter);
+
+      // save default adapter
+      this._defaultAdapter = newAdapter;
+
+      if (oldAdapter) {
+        // unwatch event since the old adapter is no longer usefull
+        this._unwatchOnhfpstatuschanged(oldAdapter);
```

Comment:
As the handlers for the status changes of the profiles are the same (note that the only difference is the name of the profile), I would suggest to iterate all the profiles defined in `Profiles` to register/unregister the handlers.  The same thing also applies to `_updateConnectionStatus` and all other parts related to profiles.
By doing this we can support newly added profiles just by adding new profile names in `Profiles` without changing other parts.